src/Microsoft.ML.StandardLearners/Standard/LinearClassificationTrainer.cs

```diff
@@ -1496,21 +1501,21 @@ protected override BinaryPredictionTransformer<TScalarPredictor> MakeTransformer
     }
 
     public sealed class StochasticGradientDescentClassificationTrainer :
-        LinearTrainerBase<TScalarPredictor>
+        LinearTrainerBase<BinaryPredictionTransformer<TScalarPredictor>, TScalarPredictor>
     {
-        public const string LoadNameValue = "BinarySGD";
-        public const string UserNameValue = "Hogwild SGD (binary)";
-        public const string ShortName = "HogwildSGD";
+        internal const string LoadNameValue = "BinarySGD";
+        internal const string UserNameValue = "Hogwild SGD (binary)";
+        internal const string ShortName = "HogwildSGD";
 
         public sealed class Arguments : LearnerInputBaseWithWeight
         {
             [Argument(ArgumentType.Multiple, HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
             public ISupportClassificationLossFactory LossFunction = new LogLossFactory();
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "L2 regularizer constant", ShortName = "l2", SortOrder = 50)]
-            [TGUI(Label = "L2 Regularizer Constant", SuggestedSweeps = "1e-7,5e-7,1e-6,5e-6,1e-5")]
+            [Argument(ArgumentType.AtMostOnce, HelpText = "L2 Regularization constant", ShortName = "l2", SortOrder = 50)]
+            [TGUI(Label = "L2 Regularization Constant", SuggestedSweeps = "1e-7,5e-7,1e-6,5e-6,1e-5")]
             [TlcModule.SweepableDiscreteParam("L2Const", new object[] { 1e-7f, 5e-7f, 1e-6f, 5e-6f, 1e-5f })]
-            public float L2Const = (float)1e-6;
```

Comment:
>L2Const [](start = 25, length = 7)

double -check with @tomfnley on this change. Think we tend to keep the names same as the names of the variables in the respective papers.

Reply:
Ok will do, thank you!

---
In reply to: [222460420](https://github.com/dotnet/machinelearning/pull/1134#discussion_r222460420) [](ancestors = 222460420)

Reply:
Checked with Tom and he is fine with the name change, as it seems commonly used in the rest of the codebase.

---
In reply to: [222463578](https://github.com/dotnet/machinelearning/pull/1134#discussion_r222463578) [](ancestors = 222463578,222460420)